core/domain/opportunity_domain.py

```diff
@@ -0,0 +1,165 @@
+        self.chapter_title = chapter_title
+        self.content_count = content_count
+        self.incomplete_translation_languages = incomplete_translation_languages
+        self.translation_counts = translation_counts
+        self.need_voice_artist_in_languages = need_voice_artist_in_languages
+        self.assigned_voice_artist_in_languages = (
+            assigned_voice_artist_in_languages)
+        self.validate()
+
+    @classmethod
+    def from_dict(cls, exploration_opportunity_summary_dict):
+        """Return a ExplorationOpportunitySummary domain object from a dict.
+
+        Args:
+            exploration_opportunity_summary_dict: dict. The dict representation
+                of ExplorationOpportunitySummary object.
+
+        Returns:
+            ExplorationOpportunitySummary. The corresponding
+                ExplorationOpportunitySummary domain object.
+        """
+        return cls(
+            exploration_opportunity_summary_dict['id'],
+            exploration_opportunity_summary_dict['topic_id'],
+            exploration_opportunity_summary_dict['topic_name'],
+            exploration_opportunity_summary_dict['story_id'],
+            exploration_opportunity_summary_dict['story_title'],
+            exploration_opportunity_summary_dict['chapter_title'],
+            exploration_opportunity_summary_dict['content_count'],
+            exploration_opportunity_summary_dict[
+                'incomplete_translation_languages'],
+            exploration_opportunity_summary_dict['translation_counts'],
+            exploration_opportunity_summary_dict[
+                'need_voice_artist_in_languages'],
+            exploration_opportunity_summary_dict[
+                'assigned_voice_artist_in_languages'])
+
+
+    def validate(self):
+        """Validates various properties of the object.
+
+        Raises:
+            ValidationError: One or more attributes of the object are invalid.
+        """
+        if not isinstance(self.topic_id, basestring):
+            raise utils.ValidationError(
+                'Expected topic_id to be a string, received %s' % self.topic_id)
+        if not isinstance(self.topic_name, basestring):
+            raise utils.ValidationError(
+                'Expected topic_name to be a string, received %s' %
+                self.topic_name)
+        if not isinstance(self.story_id, basestring):
+            raise utils.ValidationError(
+                'Expected story_id to be a string, received %s' % self.story_id)
+        if not isinstance(self.story_title, basestring):
+            raise utils.ValidationError(
+                'Expected story_title to be a string, received %s' %
+                self.story_title)
+        if not isinstance(self.chapter_title, basestring):
+            raise utils.ValidationError(
+                'Expected chapter_title to be a string, received %s' %
+                self.chapter_title)
+        if not isinstance(self.content_count, int):
```

Comment:
All int count fields should also have a check that they are greater than or equal to zero.

Reply:
Done!